LoginApplication/app/src/main/java/com/example/webonise/loginapplication/SignUpActivity.java

```diff
@@ -0,0 +1,172 @@
+        editConfirmPassword=(EditText)findViewById(R.id.editConfirmPassword);
+        editAddress=(EditText)findViewById(R.id.editAddress);
+        editanswer=(EditText)findViewById(R.id.editanswer);
+
+        radioButtonMale=(RadioButton)findViewById(R.id.radioButtonMale);
+        radioButtonFemale=(RadioButton)findViewById(R.id.radioButtonFemale);
+
+        btnLogin=(Button)findViewById(R.id.btnLogin);
+        btnCancel=(Button)findViewById(R.id.btnCancel);
+
+        spinner.setOnItemSelectedListener(this);
+        btnLogin.setOnClickListener(this);
+        btnCancel.setOnClickListener(this);
+
+        List<String> itemSpinner=new ArrayList<>();
+        itemSpinner.add("What is your nick name?");
+        itemSpinner.add("Who is your favourite actor?");
+        itemSpinner.add("Which school you went?");
+        itemSpinner.add("First pet you had?");
+        // Creating adapter for spinner
+        ArrayAdapter<String> dataAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_spinner_item, itemSpinner);
+        // Drop down layout style - list view with radio button
+        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        // attaching data adapter to spinner
+        spinner.setAdapter(dataAdapter);
+    }
+
+    @Override
+    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        String item = parent.getItemAtPosition(position).toString();
+    }
+
+    @Override
+    public void onNothingSelected(AdapterView<?> parent) {
+
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()){
+            case R.id.btnLogin:
+                validatefName();
+                validatelName();
+                validateContact();
+                validateEmail();
+                validatePassword();
+                validateAddress();
+//                toHomePage();
+                break;
+            case R.id.btnCancel:
+                System.exit(0);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void validatefName() {
+        if (editFirstName.getText().toString().trim().isEmpty()){
+            toast=Toast.makeText(getApplicationContext(),"Enter First Name",Toast.LENGTH_SHORT);
+            toast.show();
+        }else {
+            String fname=editFirstName.getText().toString();
+        }
+    }
+
+    private void validatelName() {
+        if (editLastName.getText().toString().trim().isEmpty()){
+            toast=Toast.makeText(getApplicationContext(),"Enter Last Name",Toast.LENGTH_SHORT);
+            toast.show();
+        }else {
+            String lname=editLastName.getText().toString();
+        }
+    }
+
+    private void validateContact() {
+        if (editContact.getText().toString().trim().isEmpty()){
+            toast=Toast.makeText(getApplicationContext(),"Enter Contact",Toast.LENGTH_SHORT);
```

Comment:
No need to catch toast object and call show method. You can directly call show method.  Moreover logic is also not properly designed.